File: ArrayAndString/167. Two Sum II - Input Array Is Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        if(len(numbers) == 2): return [1,2]

        #two pointer solution, but there's probably a binary search solution too.
        start = 0
        end = len(numbers)-1
        result = numbers[start]+numbers[end]
        while(result != target):
            if(result > target):
                end -= 1
            else:
                start +=1
            result = numbers[start]+numbers[end]
        return [start+1,end+1]
        

        

File: ArrayAndString/167. Two Sum II - Input Array Is Sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        if(numbers.length == 2){
            int[] quick = {1,2};
            return quick;
        } 

        //two pointer solution, but there's probably a binary search solution too.
        int start = 0, end = numbers.length-1;
        int result = numbers[start]+numbers[end];
        while(result != target){
            if(result > target) {
                end -= 1;
            }
            else {
                start +=1;
            }
            result = numbers[start]+numbers[end];
        }
        int[] quick = {start+1,end+1};
        return quick;
    }
}